src/lib/builder/Web3FunctionBuilder.spec.ts

```diff
@@ -0,0 +1,137 @@
+      buildSchemaPath("not-existing")
+    );
+
+    expect(res.success).toBeFalsy();
+    if (res.success === false) {
+      expect(
+        res.error.message.includes("Missing Web3Function schema")
+      ).toBeTruthy();
+    }
+  });
+
+  test("should fail when input path does not have schema.json", async () => {
+    const res = await Web3FunctionBuilder.build(buildSchemaPath("no-schema"));
+
+    expect(res.success).toBeFalsy();
+    if (res.success === false) {
+      expect(
+        res.error.message.includes("Missing Web3Function schema")
+      ).toBeTruthy();
+    }
+  });
+
+  test("should fail when schema is missing a required field", async () => {
+    const res = await Web3FunctionBuilder.build(
+      buildSchemaPath("missing-required-field")
+    );
+
+    expect(res.success).toBeFalsy();
+    if (res.success === false) {
+      expect(
+        res.error.message.includes("must have required property")
+      ).toBeTruthy();
+    }
+  });
+
+  test("should fail when schema major version does not match with the SDK version", async () => {
+    const res = await Web3FunctionBuilder.build(
+      buildSchemaPath("invalid-schema-version")
+    );
+
+    expect(res.success).toBeFalsy();
+    if (res.success === false) {
+      expect(
+        res.error.message.includes(
+          "must match the major version of the installed sdk"
+        )
+      ).toBeTruthy();
+    }
+  });
+
+  test("should fail when schema memory config is invalid", async () => {
+    const res = await Web3FunctionBuilder.build(
+      buildSchemaPath("invalid-schema-memory")
+    );
+
+    expect(res.success).toBeFalsy();
+    if (res.success === false) {
+      expect(
+        res.error.message.includes(
+          "'memory' must be equal to one of the allowed values"
+        )
+      ).toBeTruthy();
+    }
+  });
+
+  test("should fail when schema runtime config is invalid", async () => {
+    const res = await Web3FunctionBuilder.build(
+      buildSchemaPath("invalid-schema-runtime")
+    );
+
+    expect(res.success).toBeFalsy();
+    if (res.success === false) {
+      expect(
+        res.error.message.includes(
+          "'runtime' must be equal to one of the allowed values"
+        )
+      ).toBeTruthy();
+    }
+  });
+
+  test("should fail when schema timeout is invalid", async () => {
+    const res = await Web3FunctionBuilder.build(
+      buildSchemaPath("invalid-schema-timeout")
+    );
+
+    expect(res.success).toBeFalsy();
+    if (res.success === false) {
+      expect(res.error.message.includes("'timeout' must be")).toBeTruthy();
+    }
+  });
+
+  test("should fail when schema userArgs include unknown types", async () => {
+    const res = await Web3FunctionBuilder.build(
+      buildSchemaPath("invalid-schema-timeout")
+    );
+
+    expect(res.success).toBeFalsy();
+    if (res.success === false) {
+      expect(res.error.message.includes("'timeout' must be")).toBeTruthy();
+    }
```

Comment:
I think this one is a copy/paste of the above and is not actually checking the user args

Reply:
nice catch, fixed in 5c93b2f081209d252d59f59f2d8c50cede106f04